### PR TITLE
Created Proposal for CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by automatealchemist**
This pull request adds a CONTRIBUTING.md file to help guide new contributors who want to support the Aluminium project.

The document is written in a friendly and human tone that reflects the spirit of open collaboration. It outlines contribution steps clearly, from forking the repo to creating a pull request, while also encouraging new contributors to ask questions and learn along the way.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+Thank you for your interest in contributing to Aluminium! As an experimental AI-powered test automation framework, we are pioneering new approaches to make testing more intuitive and robust. Your expertise and passion can help shape this emerging technology.
+
+As a software tester, I have seen how the right tools can transform quality assurance. Aluminium aims to be that transformative tool by leveraging AI to simplify test interactions and assertions.
+
+## Project Understanding
+
+Before contributing, please review:
+
+- Our [README.md](https://github.com/automatealchemist/alumnium#) to understand Aluminium's vision: creating higher-level abstractions for test automation that simplify web page interactions and strengthen assertion mechanisms
```
Comment (sh3pik) [inline CONTRIBUTING.md]:
```suggestion
- Our [docs](https://alumnium.ai/docs/) and [README.md](https://github.com/alumnium-hq/alumnium) to understand Aluminium's vision: creating higher-level abstractions for test automation that simplify web page interactions and strengthen assertion mechanisms
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+- Our experimental status - we're in early development and value innovative approaches
+- The core functionality that uses natural language processing to interpret testing commands
+
+## Finding Your Contribution Opportunity
+
+- Explore the [open issues](https://github.com/yourusername/aluminium/) to find tasks matching your interests
```
Comment (sh3pik) [inline CONTRIBUTING.md]:
```suggestion
- Explore the [open issues](https://github.com/alumnium-hq/alumnium/issues) to find tasks matching your interests
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+# Fork and clone the repository
+git clone https://github.com/your-username/aluminium.git
+cd aluminium
+
+# Set up development environment
+pipx install poetry
+poetry install
+
+# Configure AI provider access as mentioned in [README.md](https://github.com/automatealchemist/alumnium#)
+```
```
Comment (sh3pik) [inline CONTRIBUTING.md:41]: I think it may be better to move it as a separate step. A contributor will need to decide which model to use.
```suggestion
### 2. Configure AI provider access

Configure access to AI providers as mentioned in [docs](https://alumnium.ai/docs/getting-started/configuration/)
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+- Be respectful and constructive in all interactions
+- Share knowledge generously - we're all learning in this emerging field
+- Value diverse perspectives - they lead to more robust solutions
+- Ask questions when unclear - clarity benefits everyone
+
+## For First-Time Contributors
+
+If you're new to open-source or AI-powered testing:
+
+1. Try running the demo and experimenting with the Alumni API
```
Comment (sh3pik) [inline CONTRIBUTING.md]:
```suggestion
1. Try running the demo and experimenting with the Alumnium API
```

**Review by sh3pik — CHANGES_REQUESTED** [review]
@automatealchemist, thanks a lot for the pull request! Appreciate your time and work! 
I left some small comments, we can merge the changes as soon as they are resolved 🙏

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
+# Contributing to Aluminium
```
Comment (sh3pik) [inline CONTRIBUTING.md]:
```suggestion
# Contributing to Alumnium
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+## Welcome to the Future of Test Automation!
+
+Thank you for your interest in contributing to Aluminium! As an experimental AI-powered test automation framework, we are pioneering new approaches to make testing more intuitive and robust. Your expertise and passion can help shape this emerging technology.
```
Comment (sh3pik) [inline CONTRIBUTING.md]:
```suggestion
Thank you for your interest in contributing to Alumnium! As an experimental AI-powered test automation framework, we are pioneering new approaches to make testing more intuitive and robust. Your expertise and passion can help shape this emerging technology.
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+
+As a software tester, I have seen how the right tools can transform quality assurance. Aluminium aims to be that transformative tool by leveraging AI to simplify test interactions and assertions.
```
Comment (sh3pik) [inline CONTRIBUTING.md]:
```suggestion
As a software tester, I have seen how the right tools can transform quality assurance. Alumnium aims to be that transformative tool by leveraging AI to simplify test interactions and assertions.
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+
+## Project Understanding
+
+Before contributing, please review:
+
+- Our [docs](https://alumnium.ai/docs/) and [README.md](https://github.com/alumnium-hq/alumnium) to understand Aluminium's vision: creating higher-level abstractions for test automation that simplify web page interactions and strengthen assertion mechanisms
```
Comment (sh3pik) [inline CONTRIBUTING.md]:
```suggestion
- Our [docs](https://alumnium.ai/docs/) and [README.md](https://github.com/alumnium-hq/alumnium) to understand Alumnium's vision: creating higher-level abstractions for test automation that simplify web page interactions and strengthen assertion mechanisms
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+
+## Finding Your Contribution Opportunity
+
+- Explore the [open issues](https://github.com/alumnium-hq/alumnium/issues) to find tasks matching your interests
+- We will be glad if you help us with :
+  - Improving test coverage for edge cases  
+  - Enhancing documentation and examples  
+  - Exploring more natural language prompts for test generation  
+  - Reporting usability issues or unexpected test behavior  
+  - Creating sample projects using Aluminium
```
Comment (sh3pik) [inline CONTRIBUTING.md]:
```suggestion
  - Creating sample projects using Alumnium
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+
+## Contribution Workflow
+
+### 1. Environment Setup
+
+```bash
+# Fork and clone the repository
+git clone https://github.com/your-username/aluminium.git
+cd aluminium
```
Comment (sh3pik) [inline CONTRIBUTING.md]:
```suggestion
git clone https://github.com/your-username/alumnium.git
cd alumnium
```

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+
+```
+
+### 2. Configure AI provider access
+Configure access to AI providers as mentioned in [docs](https://alumnium.ai/docs/getting-started/configuration/)
+
+
+### 3. Development Guidelines
+
+When working on Aluminium:
```
Comment (sh3pik) [inline CONTRIBUTING.md]:
```suggestion
When working on Alumnium:
```

**Review by sh3pik — COMMENTED** [review]
@automatealchemist Thanks for the prompt update! Just a couple of more comments and good to go

**Review by automatealchemist — COMMENTED** [review]
Hi @sh3pik 

Apologies for the confusion. I misread the project name as "aluminium", thinking it was a play on the metal similar to Selenium. Based on that assumption, I drafted the file using that spelling.

I have now corrected the text to use the accurate project name "alumnium" consistently across the file.

Please have a look and let me know if any further changes are needed. Happy to update as required.

Thanks,
Ujjwal

**Review by sh3pik — APPROVED** [review]
